newdle/client/src/util/hooks.js

```diff
@@ -0,0 +1,28 @@
+import {useState} from 'react';
+
+/**
+ * This hook handles the possible states regarding backend communication.
+ *
+ * @param {Function} func - the client function which will be invoked
+ */
+export function useClient(func) {
```

Comment:
what about putting this as an extra export into `client.js`?

Reply:
Thought of it too, I don't have a strong opinion about it.